src/elements/element.arc.js

```diff
@@ -19,15 +22,137 @@ function clipArc(ctx, element) {
+    oe = numberOrZero(value.outerEnd);
+    is = numberOrZero(value.innerStart);
+    ie = numberOrZero(value.innerEnd);
+  } else {
+    os = oe = is = ie = numberOrZero(value);
+  }
+
+  return {
+    outerStart: os,
+    outerEnd: oe,
+    innerStart: is,
+    innerEnd: ie
+  };
+}
+
+/**
+ * Parse border radius from the provided options
+ * @param {ArcElement} arc
+ * @param {number} innerRadius
+ * @param {number} outerRadius
+ * @param {number} angleDelta Arc circumference in radians
+ * @returns
+ */
+function parseBorderRadius(arc, innerRadius, outerRadius, angleDelta) {
+  const o = toRadiusCorners(arc.options.borderRadius);
+  const halfThickness = (outerRadius - innerRadius) / 2;
+  const innerLimit = Math.min(halfThickness, angleDelta * innerRadius / 2);
+
+  // Outer limits are complicated. We want to compute the available angular distance at
+  // a radius of outerRadius - borderRadius because for small angular distances, this term limits.
+  // We compute at r = outerRadius - borderRadius because this circle defines the center of the border corners.
+  //
+  // If the borderRadius is large, that value can become negative.
+  // This causes the outer borders to lose their radius entirely, which is rather unexpected. To solve that, if borderRadius > outerRadius
+  // we know that the thickness term will dominate and compute the limits at that point
+  const computeOuterLimit = (val) => {
+    const outerArcLimit = (outerRadius - Math.min(halfThickness, val)) * angleDelta / 2;
+    return _limitValue(val, 0, Math.min(halfThickness, outerArcLimit));
+  };
+
+  return {
+    outerStart: computeOuterLimit(o.outerStart),
+    outerEnd: computeOuterLimit(o.outerEnd),
+    innerStart: _limitValue(o.innerStart, 0, innerLimit),
+    innerEnd: _limitValue(o.innerEnd, 0, innerLimit),
+  };
+}
+
+/**
+ * Convert (r, 𝜃) to (x, y)
+ * @param {number} r Radius from center point
+ * @param {number} theta Angle in radians
+ * @param {number} x Center X coordinate
+ * @param {number} y Center Y coordinate
+ * @returns {{ x: number; y: number }} Rectangular coordinate point
+ */
+function rThetaToXY(r, theta, x, y) {
+  return {
+    x: x + r * Math.cos(theta),
+    y: y + r * Math.sin(theta),
+  };
+}
+
+
+/**
+ * Path the arc, respecting the border radius
+ *
+ * 8 points of interest exist around the arc segment.
+ * These points define the intersection of the arc edges and the corners.
+ *
+ *    1---------2
+ *   /           \
+ *   8           3
+ *   |           |
+ *   |           |
+ *   7           4
+ *   \           /
+ *    6---------5
+ * @param {CanvasRenderingContext2D} ctx
+ * @param {ArcElement} element
+ */
 function pathArc(ctx, element) {
   const {x, y, startAngle, endAngle, pixelMargin} = element;
   const outerRadius = Math.max(element.outerRadius - pixelMargin, 0);
   const innerRadius = element.innerRadius + pixelMargin;
-
+  const borderRadius = parseBorderRadius(element, innerRadius, outerRadius, endAngle - startAngle);
```

Comment:
Destructuring should save some bytes:
```suggestion
  const {outerStart, outerEnd, innerEnd, innerStart} = parseBorderRadius(element, innerRadius, outerRadius, endAngle - startAngle);
```